hank-core/src/main/java/com/liveramp/hank/coordinator/Hosts.java

```diff
@@ -99,6 +101,30 @@ public static boolean isUpToDate(Host host, DomainGroup domainGroup) throws IOEx
     return true;
   }
 
+  // Return true if all partitions assigned to that host for the given domain are at the correct version, and
+  // there are no deletable partitions.
+  public static boolean isUpToDate(Host host, DomainGroup domainGroup, Domain domain) throws IOException {
+
+    if (domain == null || domain.getVersions() == null) {
+      return false;
+    }
+
+    Set<DomainAndVersion> domainAndVersion = new HashSet<>();
+    for (DomainAndVersion d : domainGroup.getDomainVersions()) {
+      if (d.getDomain().equals(domain)) domainAndVersion.add(d);
+    }
+
```

Comment:
I think after this point you can forward to isUpToDateOrMoreRecent  with the domainAndVersion(s)

Reply:
Ah, good catch.